rocky/rocky/scheduler.py

```diff
@@ -223,31 +225,28 @@ def list_tasks(
         **kwargs,
     ) -> PaginatedTasksResponse:
         try:
-            kwargs = {k: v for k, v in kwargs.items() if v is not None}  # filter Nones from kwargs
-            res = self._client.get("/tasks", params=kwargs)
+            filter_key = "filters"
+            params = {k: v for k, v in kwargs.items() if v is not None if k != filter_key}  # filter Nones from kwargs
+            endpoint = "/tasks"
+            res = self._client.post(endpoint, params=params, json=kwargs.get(filter_key, None))
             return PaginatedTasksResponse.model_validate_json(res.content)
         except ValidationError:
             raise SchedulerValidationError(extra_message=_("Task list: "))
         except ConnectError:
             raise SchedulerConnectError(extra_message=_("Task list: "))
 
     def get_task_details(self, task_id: str) -> Task:
-        try:
-            res = self._client.get(f"/tasks/{task_id}")
-            res.raise_for_status()
-            task_details = Task.model_validate_json(res.content)
+        task_details = Task.model_validate_json(self._get(f"/tasks/{task_id}", "content"))
```

Comment:
Seeing that we will check *if* the organization of the returned job is the same as the now optionally set self.organization we should probably just don't do anything when there is not self.organization defined.